src/blockchain/log-processors/market-created.ts

```diff
@@ -44,111 +68,109 @@ export function processMarketCreatedLog(db: Knex, augur: Augur, log: FormattedEv
-          marketCreatorMailbox:       onContractData.marketCreatorMailbox,
-          marketCreatorMailboxOwner:  log.marketCreator,
-          initialReportSize:          null,
-          reportingFeeRate:           convertDivisorToRate(onContractData.reportingFeeDivisor!, 10),
-          marketCreatorFeesBalance:   "0",
-          volume:                     "0",
-          sharesOutstanding:          "0",
-          openInterest:               "0",
-          validityBondSize:           onContractData.validityBondAttoeth,
-          forking:                    0,
-          needsMigration:             0,
-          needsDisavowal:             0,
-          finalizationBlockNumber:    null,
+          feeWindow: onContractData.feeWindow,
+          endTime: parseInt(onContractData.endTime!, 10),
+          designatedReporter: onContractData.designatedReporter,
+          designatedReportStake: convertFixedPointToDecimal(designatedReportStakeRow.balance, WEI_PER_ETHER),
+          numTicks: onContractData.numTicks,
+          marketCreatorFeeRate: convertDivisorToRate(onContractData.marketCreatorSettlementFeeDivisor!, 10),
+          marketCreatorMailbox: onContractData.marketCreatorMailbox,
+          marketCreatorMailboxOwner: log.marketCreator,
+          initialReportSize: null,
+          reportingFeeRate: convertDivisorToRate(onContractData.reportingFeeDivisor!, 10),
+          marketCreatorFeesBalance: "0",
+          volume: "0",
+          sharesOutstanding: "0",
+          openInterest: "0",
+          validityBondSize: onContractData.validityBondAttoeth,
+          forking: 0,
+          needsMigration: 0,
+          needsDisavowal: 0,
+          finalizationBlockNumber: null,
         };
         const outcomesDataToInsert: Partial<OutcomesRow<string>> = formatBigNumberAsFixed<Partial<OutcomesRow<BigNumber>>, Partial<OutcomesRow<string>>>({
           marketId: log.market,
-          price: new BigNumber(log.minPrice, 10).plus(new BigNumber(log.maxPrice, 10)).dividedBy(new BigNumber(numOutcomes, 10)),
+          price: new BigNumber(log.minPrice, 10).plus(new BigNumber(log.maxPrice, 10)).dividedBy(new BigNumber(onContractData.getOutcomes.numOutcomes, 10)),
           volume: ZERO,
           shareVolume: ZERO,
         });
         const tokensDataToInsert: Partial<TokensRow> = {
           marketId: log.market,
           symbol: "shares",
         };
-        const shareTokens = new Array(numOutcomes);
-        forEachOf(shareTokens, (_: null, outcome: number, nextOutcome: ErrorCallback): void => {
-          augur.api.Market.getShareToken(Object.assign({ _outcome: outcome }, marketPayload), (err: Error|null, shareToken?: Address): void => {
-            if (err) return nextOutcome(err);
-            shareTokens[outcome] = shareToken;
-            nextOutcome(null);
-          });
-        }, (err: Error|null): void => {
+
+        series([
```

Comment:
Looks like this `series` should be able to be pulled out into just a sequential list of awaits pretty easily, add to the list?